core/utils.ts

```diff
@@ -219,3 +219,12 @@ export function checkExcessProperties<T>(
     );
   }
 }
+
+export function throwIfTrailingSemicolonInQuery(session: Session, query: string) {
```

Comment:
nit: IMO this should not be called `throw`, since it doesn't.

`validateQueryString`?